zipline/utils/pandas_utils.py

```diff
@@ -91,6 +91,39 @@ def mask_between_time(dts, start, end, include_start=True, include_end=True):
     )
 
 
+def find_in_sorted_index(dts, dt):
+    """
+    Find the index of ``dt`` in ``dts``.
+
+    This function should be used instead of `dts.get_loc(dt)` if the index is
+    large enough that we don't want to initialize a hash table in ``dts``. In
+    particular, this should always be used on minutely trading calendars.
+
+    Parameters
+    ----------
+    dts : pd.DatetimeIndex
+        Index in which to look up ``dt``. **Must be sorted**.
+    dt : pd.Timestamp
+        ``dt`` to be looked up.
+
+    Returns
+    -------
+    ix : int
+        Integer index such that dts[ix] == dt.
+
+    Raises
+    ------
+    KeyError
+        If dt is not in ``dts``.
+    """
+    ix = dts.searchsorted(dt)
+    if dts[ix] != dt:
+        raise KeyError(
+            "{0} is not in calendar [{1} ... {2}]".format(dt, dts[0], dts[-1])
```

Comment:
It would be a degenerate case, but the logging here fails if `dts` has no length.
Raising `"{0} is not in calendar {1}".format(dt, dts)` would fix that case, as well as providing extra information about the `DatetimeIndex`, e.g. the timezone, which is useful if this function is used generally, (anectdotallaly, a mismatch between timezones of the sought after `Timestamp` and the `DatetimeIndex` has been a head scratcher at least once.